File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    var arr=s.split("");
    
    let stack=[];
    let idx=0;
    for(let i=0; i<arr.length;i++){
        if(arr[i]=="("){
            stack.push(arr[i]);
        }else if(arr[i]==")"){
            if(stack.length==0){
                answer=false;
                break;
            }
            stack.pop();
        }
    }
    if(stack.length!=0){
        answer=false;
    }

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        Stack<Character> stack=new Stack<>();
        char[] arr=s.toCharArray();
        
        for(int i=0; i<arr.length; i++){
            char cur=arr[i];
            if(stack.isEmpty()) {
                if(cur==')'){
                    answer=false;
                    break;
                }
                stack.push(arr[i]);
                continue;
            }
            
            if(cur=='('){
                stack.push(cur);
            }else{
               if(stack.peek()=='('){
                    stack.pop();
                }else{
                    answer=false;
                    break;
                }
            }
        }

        if(!stack.isEmpty()){
            answer=false;
        }
        return answer;
    }
}